deploy/operator/00_operator_remote.clusterrole_def.yaml

```diff
@@ -47,6 +47,8 @@ rules:
   - endpoints
   - namespaces
   - services
+  - secrets
```

Comment:
This is all verbs across all namespaces for secrets. This may have been discussed before, but I don't think that creating such a wide clusterrole is a good idea from the security design perspective, and I'd avoid it even at the cost of UX.

Some alternatives off the top of my head (with various degrees of disruption):
- switch from secrets to CRDs for the specific bits we need to propagate
- only grant on the per-namespace basis (might require manual grants by the user)

Reply:
We can't predict in which namespaces users will create their ScyllaDBCluster/ScyllaDBDatacenter nor predict names of their Secrets and ConfigMaps used within our CRs. Restricting those would be a terrible UX choice and I'm fully against it.

Reply:
Discussed offline with a conclusion that we'd pave a way towards a possibility to restrict the set of namespaces that operator remotely writes to to a predefined list. But the issue mentioned in https://github.com/scylladb/scylla-operator/pull/2524#discussion_r2017409079 has uncovered yet another concern, so the resolution here is TBD based on our higher-order approach.